### ISSUE TITLE
Unable to select default template for translation editing

### ISSUE THREAD

**Issue by raidnet-ms** [PrestaShop/PrestaShop#26140]
#### Describe the bug

Choosing the template in international settings to change laguage field for the frontend is not possible anymore. you get a "null" value, no template selected and you can't edit the default template frontend language.

#### Expected behavior

As in 1.7.7.6 I'm supposed to be able to select the frontoffice template to edit language fields.

#### Steps to Reproduce

1. Go to 'international'
2. Click on 'translate'
3. Scroll down to 'frontoffice translation'
4. See error: 'select your theme' has nothing in the dropdown selection

**Screenshots**

If applicable, add screenshots or screen recordings to help explain your problem.
<img width="875" alt="nothemefield" src="https://user-images.githubusercontent.com/22340972/136387445-d2e17e95-6a74-4d81-83a7-5535faba865e.png">

#### Additional information

* PrestaShop version: 1.7.8.0
* PHP version: 7.3

**Comment by hibatallahAouadni** [comment]
Hello @raidnet-ms 

I didn't manage to reproduce your issue with PS1778, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/136414538-7765735a-743e-41be-ad52-e0fd3d85025f.png)

---
![image](https://user-images.githubusercontent.com/23308427/136414361-9ff128a7-f66b-4844-9f2b-6b9b640f7dba.png)

Please check and feedback.
Thanks!

**Comment by raidnet-ms** [comment]
I updated from 1.7.7.6 to 1.7.8.0 with 1click update.
Maybe it is an update fail? I also find a strange behaviour with the template (search module is rendering bad, fonts are different

**Comment by raidnet-ms** [comment]
Something with classic template happened during update.
Is there a way to install it again? So i might check if it was an update error.

**Comment by raidnet-ms** [comment]
The problem occours even if i compile it again, if i delete and disable cache, and if i enable debug mode disabling overrides and non native modules.

**EDIT: problem occours with version 1.7.8, not 1.7.7.8, sorry for typo.**

**Comment by raidnet-ms** [comment]
Could you please change the version flag for this bug report with 1.7.8?

Thank you

**Comment by khouloudbelguith** [comment]
Hi @raidnet-ms,

I did not manage to reproduce the issue with PS1780 after an upgrade from PS1777 using the 1-click upgrade module
I attached a screen record
https://watch.screencastify.com/v/bB2MgwPBBvAAOBN3PeC0
Could you please try to configure your smarty by disabling the cache, set the option "Template compilation" to Force compilation, and clear the cache.
Also, try to open your shop in an incognito browser.
You need to force refresh the page, by pressing CTRL+F5 on Windows / Linux, Cmd+R on Mac OS while you are on this page.
Thanks to check and feedback.

**Comment by raidnet-ms** [comment]
Already tryied everything before opening this post.

Is there a way to install again the default template?

**Comment by kpodemski** [comment]
I don't think it's related to the theme installation. It seems that you have some modifications that disable the first option in this select field. Is there any chance of it? In 1.7.8 there are no longer translations for "classic", "classic" translations = "core translations".

**Comment by raidnet-ms** [comment]
I have not any modification on this template, neighter I am using any 3th party component. It's a very simple installation.

I updated from 1.7.7.6 so I missed the 1.7.7.7 update maybe is related to this?

Where I can find "core translations?" I need to translate some "registration" fields that previously were integrated into "frontend->classic" theme.

**Comment by raidnet-ms** [comment]
Without template selection from frontend I can't find SIRET in the registration form to translate it properly.

Anyway seems there's also another problem related with this field because the italian corresponding for siret has 16 characters, not 14 and while there's no limit to the field during the registration, the form will save only first 14 characters.

**Comment by raidnet-ms** [comment]
Ok i solved the translation in this way, I select "nucleo" instead of "classic". "Nucleo" is a really bad translation for "core" in italian, anyway, after goin into "core translation" i found that the field was moved to "fields". Here I found a non corresponding translation: in the frontend there was "VAT/ID" while in the backend translation there was only "ID" . I made a reset for the field and then i was able to see the correct name into the frontend.

Well if that breaking change was more detailed into the changelog I would solve this withtout bothering you with this issue.

The truncating character in SIRET FIELD is still happening. shall I open another Issue?

**Comment by hibatallahAouadni** [comment]
Hello @raidnet-ms 

> Well if that breaking change was more detailed into the changelog I would solve this withtout bothering you with this issue.

:laughing: well at last you solved your issue :tada: 

> The truncating character in SIRET FIELD is still happening. shall I open another Issue?

No, you don't have, just provide clear and concise steps to reproduce the issue and if you could provide a screen record, we will be thankful :pray: 
Please check and feedback.

Thanks!

**Comment by raidnet-ms** [comment]
Sure I'll try to be as clear as possible. I have **b2b** option activated.

In the classic template, when an user is trying to register, the default fields prompted for registration are:
![image](https://user-images.githubusercontent.com/22340972/136828542-e10d77b8-f0eb-48f9-a41e-14167e060610.png)
Where CODICE FISCALE (default translation for this default field) is supposed to stand for DNI but I think it's wrongly related to SIRET (french VAT).

Anyway, if I put an italialian ID (it has 16 characters) prestashop will save only first 14 characters in the SIRET field
![image](https://user-images.githubusercontent.com/22340972/136829860-04773ddc-6144-4c28-a36f-0d8215a3baae.png)

Moreover If we go in shipping details we get another CODICE FISCALE field (DNI), but this time is empty
![image](https://user-images.githubusercontent.com/22340972/136830507-665abf49-f9b1-4b2c-a6c1-9d0bc1be2335.png)

If i remove DNI field in "international->location->nation" the field will be removed from the "address view" but not from the profile and registration view".
![image](https://user-images.githubusercontent.com/22340972/136830728-8b057943-7e7a-4cfa-ab30-fc1c738aaa80.png)

**In conclusion**
Seems the french VAT (SIRET) was wrongly translated to DNI in italian.  In any case I'm unable to remove the wrong 14 characters field (SIRET) from the user detail/registration form and change it with italian VAT number.

**Comment by raidnet-ms** [comment]
Identification number field is the term used for the frontend while into the backend is translated as SIRET (and has 12 characters limit).

French siret is something related only to companies, while identification number (or Codice fiscale) in Italy is something may be related to the person and they also differ from character number.

So I don't understand what is this field you call identification number for frontend and siret for backend.

**Comment by florine2623** [comment]
Ping @Julievrz ,

> French siret is something related only to companies, while identification number (or Codice fiscale) in Italy is something may be related to the person and they also differ from character number.

What do you think about that ? Are those 2 separate information ? Should those fields be differentiated ?

**Comment by raidnet-ms** [comment]
Another possibility could be translating this field in "REA" but I suggest this way if this field appears only with b2b option enabled.

If the field appears also with b2b disabled then may be inappropriate.

REA code in Italy allows to check if the company is formally registered and if has a VAT number.

So this solution would suit the case where the ecommerce is b2b or both b2c and b2b so the ecommerce owner could  check with REA field if the registered user is a company and may let him access the proper customer group with discount prices or not.

Knowing if the registered user is a real company is mandatory before their first order or you could not allow them to see b2b reserved prices.

VAT field is the only other value that could let you know if the user is a company, but it will be asked only during the first order when the user will fill address fields (in case that you allow orders for not registered users). Unfortunately if you have a b2b the user must see reserved prices before placing the order and you won't know if that user is a company untill you will get Vat or rea code.

Actually what is sure is that you are calling that user field "DNI" in the frontend while Siret in the backend and the field will save only 14 characters while Italian dni (codice fiscale) has 16 characters. Moreover you have another dni field in address fields so actually is duplicated (one in user field other in address).

That's why renaming REA seems to me the most clever and simple solution but only you know if that field has also other use.

**Comment by Julievrz** [comment]
@raidnet-ms Can you tell us how many characters does the REA code have?

@marionf or @MatShir, what do you think? In my opinion, the SIRET code is expected here (not the Codice Fiscale which seems to be individual information) because the SIRET field only appears in B2B mode, just under the company's name, and is indeed 14 characters long. So in my opinion, it should be SIRET in the back office **and** in the front office. It might be a translation issue. What should translators do? In my opinion, there are 2 solutions:

1. Write "SIRET (French registration number)". In that case, I think we should add a help text explaining briefly what it is and how many characters are allowed. And customers that are not French wouldn't have to fill in the field.
2. Write "Company registration number" (It's more precise than "Identification number" and find an equivalent of SIRET for their country. For example, the "REA" for Italy as @raidnet-ms is suggesting. But in that case, it's weird to restrict the number of characters allowed, as it might differ depending on the code of each country. 


In any case, if the 14 characters limit is kept, an error message should be displayed in case this limit is exceeded. Because right now, only 14 characters are saved and there is no error message and I think this can be considered a bug.

**Comment by raidnet-ms** [comment]
> @raidnet-ms Can you tell us how many characters does the REA code have?
> 
>
> 
> 

The REA number consists of a progressive decimal number and is always associated with a coding, which precedes the progressive number, of the province / Chamber of Commerce that assigned it and, in practice, also includes this coding (e.g. RM-123456).

If you consider aslo the dash, it's 9 characters so it would fit the 14 limit of actual siret field.
Yes we could solve this with a correct translation.

REA is the closest thing to french SIRET.

If the field appears only in b2b mode, then also the logic is good because with that information the site owner can define if the customer is suitable for b2b prices or b2c, and after being approved, the customer can buy with the correct price and place his first order registering his vat in the address form (in case you allow orders without registration they would fill vat field only in the end of the process so REA it is a must have on user registration field).


- I think that was a simple translation bug.
- The other bug may be related to the absence of an alert if you fill more then 14 characters.
- a conditional field to prompt siret/rea field only when company name is filled, would be cool 😎

**Comment by raidnet-ms** [comment]
Where can I find the meaning of these labels?
Tnx

**Comment by Julievrz** [comment]
> Where can I find the meaning of these labels? Tnx

You can have a look here: https://github.com/PrestaShop/PrestaShop/labels ☺️

**Comment by raidnet-ms** [comment]
Oh thank you Julievrz,
From the app I wasn't able to read the overlay of labels. 

TBS stands for: "Issue to be specified", can I do anything else to be more specific?

**Comment by hibatallahAouadni** [comment]
Hello @raidnet-ms 

Usually we use `TBS` label with `Waiting for PM` or `Waiting for Dev`, it means that the issue need to be specified by Product Team or Developer Team.

Thanks!

**Comment by raidnet-ms** [comment]
awe latest update overwrited my edit and it is bugged again. i tought this was going to be achieved with 1.7.8.1.
